dbms/src/Flash/Planner/PhysicalPlanNode.h

```diff
@@ -54,7 +58,11 @@ class PhysicalPlanNode
 
     virtual size_t childrenSize() const = 0;
 
-    virtual void transform(DAGPipeline & pipeline, Context & context, size_t max_streams);
+    virtual void buildBlockInputStream(DAGPipeline & pipeline, Context & context, size_t max_streams);
+
+    virtual void buildPipelineExec(PipelineExecGroupBuilder & /*group_builder*/, Context & /*context*/, size_t /*concurrency*/);
+
+    virtual void buildPipeline(PipelineBuilder & pipeline_builder, const PipelinePtr & pipeline);
```

Comment:
Is it possible to **ONLY** pass in a `PipelineBuilder` for this method? Now the signature seems a little weird.

Reply:
It hard to hold the cur_pipeline in PipelineBuilder
consider the pipeline dag as follow
```
            JoinProbe
	           |
	+----------+
	|          |
	|          |
  JoinBuild  FinalAgg
    |          |
	|          |
   scan      PreAgg
               |
			   |
			  scan

pipelineA: scan --> JoinBuild
pipelineB: scan --> PreAgg
pipelineC: FinalAgg --> JoinProbe
```       
If we hold the cur_pipeline in PipelineBuilder, it is hard to create the dependencies of three pipelines.

Reply:
Maybe we can refine function like `void buildPipelines(const PipelinePtr & cur_pipeline, PipelineBuildState & state);`